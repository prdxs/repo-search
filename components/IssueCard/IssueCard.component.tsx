import React from 'react';
import clsx from 'clsx';
import styled from 'styled-components';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';

import { IIssueCardProps } from './IssueCard.typings';

const IssueCard: React.FC<IIssueCardProps> = ({ className, style, title }) => (
  <Card
    className={clsx('IssueCard-root', className)}
    style={style}
    data-testid="IssueCard-root"
  >
    <CardHeader title={<Typography variant="h6">{title}</Typography>} />
  </Card>
);

const StyledIssueCard = styled(IssueCard)`
  &.IssueCard-root {
    position: relative;
    min-width: 300px;
  }
`;

export default StyledIssueCard;
