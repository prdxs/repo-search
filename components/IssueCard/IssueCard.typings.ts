import { IComponentProps } from '@/typings/common';

export interface IIssueCardProps extends IComponentProps {
  title: string;
  link: string;
  state: 'open' | 'closed';
}
