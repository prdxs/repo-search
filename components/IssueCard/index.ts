export * from './IssueCard.typings';
export { default } from './IssueCard.component';
