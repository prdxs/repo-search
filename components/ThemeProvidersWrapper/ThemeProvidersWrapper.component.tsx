import React from 'react';
import { ThemeProvider } from 'styled-components';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';

import theme from '@/theme';
import { IComponentProps } from '@/typings/common';

const ThemeProvidersWrapper: React.FC<IComponentProps> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  </ThemeProvider>
);

export default ThemeProvidersWrapper;
