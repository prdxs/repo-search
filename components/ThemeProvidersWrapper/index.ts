export { default } from './ThemeProvidersWrapper.component';
