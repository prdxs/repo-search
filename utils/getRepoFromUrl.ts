const getRepoFromUrl = (url: string): { owner: string; repo: string } => {
  const parts = url.split('/');
  const len = parts.length;
  const owner = parts[len - 2];
  const repo = parts[len - 1];

  return { owner, repo };
};

export default getRepoFromUrl;
