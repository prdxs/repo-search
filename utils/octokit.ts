import { Octokit } from '@octokit/rest';

const octokit = new Octokit({
  baseUrl: 'https://api.github.com',
  auth: process.env.NEXT_PUBLIC_GH_TOKEN,
});

export default octokit;
