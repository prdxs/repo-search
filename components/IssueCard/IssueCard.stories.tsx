import React from 'react';
import { withKnobs, text, select } from '@storybook/addon-knobs';

import StorybookWrapper from '@/components/StorybookWrapper';
import IssueCard from './IssueCard.component';
import { EIssueState } from './IssueCard.typings';

export default {
  title: 'IssueCard',
  component: IssueCard,
  decorators: [withKnobs],
};

export const Default: React.FC = () => {
  return (
    <StorybookWrapper>
      <IssueCard
        title={text('name', 'bug')}
        link={text('link', 'https://github.com/facebook/react/issues/19245')}
        state={select(
          'state',
          [EIssueState.Open, EIssueState.Closed],
          EIssueState.Open
        )}
      />
    </StorybookWrapper>
  );
};
