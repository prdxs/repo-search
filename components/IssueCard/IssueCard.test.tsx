import React from 'react';
import { mount } from 'enzyme';
import Link from '@material-ui/core/Link';
import AssignmentLateIcon from '@material-ui/icons/AssignmentLate';
import AssignmentTurnedInIcon from '@material-ui/icons/AssignmentTurnedIn';

import { render } from '@/test/utils';
import ThemeProvidersWrapper from '@/components/ThemeProvidersWrapper';
import IssueCard from './IssueCard.component';

describe('IssueCard', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <IssueCard
        title="bug"
        link="https://github.com/facebook/react/issues/19245"
        state="open"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render title', () => {
    const title = '2020 is trying to kill us';
    const { queryByText } = render(
      <IssueCard
        title={title}
        link="https://github.com/facebook/react/issues/19245"
        state="open"
      />
    );

    const titleEl = queryByText(title);

    expect(titleEl).toBeInTheDocument();
  });

  it('should render link using the href provided in the link prop', () => {
    const link = 'https://github.com/facebook/react/issues/19245';
    const wrapper = mount(
      <ThemeProvidersWrapper>
        <IssueCard title="bug" link={link} state="open" />
      </ThemeProvidersWrapper>
    );

    expect(wrapper.find(Link).prop('href')).toEqual(link);
  });

  it('should render the proper icon given the status prop', () => {
    const wrapper = mount(
      <IssueCard
        title="bug"
        link="https://github.com/facebook/react/issues/19245"
        state="open"
      />
    );

    expect(wrapper.find(AssignmentLateIcon)).toHaveLength(1);
    expect(wrapper.find(AssignmentTurnedInIcon)).toHaveLength(0);

    wrapper.setProps({ state: 'closed' });

    expect(wrapper.find(AssignmentLateIcon)).toHaveLength(0);
    expect(wrapper.find(AssignmentTurnedInIcon)).toHaveLength(1);
  });
});
