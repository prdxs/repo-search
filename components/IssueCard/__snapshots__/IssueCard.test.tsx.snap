// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style="padding: 2rem;"
  >
    <div
      class="MuiPaper-root MuiCard-root IssueCard-root IssueCardcomponent__StyledIssueCard-sc-1lg5wpy-0 etfYhE MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="IssueCard-root"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-avatar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorError"
            focusable="false"
            title="Issue Open"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-6 15h-2v-2h2v2zm0-4h-2V8h2v6zm-1-9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
            />
          </svg>
        </div>
        <div
          class="MuiCardHeader-content"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
              href="https://github.com/facebook/react/issues/19245"
              target="_blank"
            >
              bug
            </a>
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
